docs/guides/building-forms.md

```diff
@@ -0,0 +1,387 @@
+  <input type="text" id="first" name="first" />
+  <label for="last">Last name:</label>
+  <input type="text" id="last" name="last" />
+  <button type="submit">Submit</button>
+</form>
+```
+
+The front-end looks like this:
+
+![html forms](https://assets.vercel.com/image/upload/c_scale,w_675/v1643009088/nextjs/guides/building-forms/html-forms.png)
+
+The HTML `<form>` tag acts as a container for different `<input>` elements like `text` field and submit `button`. Let's study each of these elements:
+
+- `action`: An attribute that specifies where the form data is sent when the form is submitted. It's generally a URL (an absolute URL or a relative URL).
+- `method`: Specifies the HTTP method, i.e., `GET` or `POST` used to send data while submitting the form.
+- `<label>`: An element that defines the label for other form elements. Label aids accessibility, especially for screen readers.
+- `<input>`: The form element that is widely used to structure the form fields. It depends significantly on the value of the `type` attribute. Input types can be `text`, `checkbox`, `email`, `radio`, and more.
+- `<button>`: It represents a clickable button that's used to submit the form data.
+
+### Form Validation
+
+A process that checks if the information provided by a user is correct or not. Form validation also ensures that the provided info is in the correct format (e.g. there's an @ in the email field). These are of two types:
+
+- **Client-side**: Validation is done in the browser
+- **Server-side**: Validation is done on the server
+
+Though both these types are equally important, for this guide, we'll focus on client-side validation only.
+
+Client-side validation is further categorized as:
+
+- **Built-in**: Uses HTML-based attributes like `required`, `type`, `minLength`, `maxLength`, `pattern` etc.
+- **JavaScript-based**: Validation that's coded with JavaScript.
+
+### Built-in Form Validation Using `required`, `type`, `minLength`, `maxLength`
+
+- `required`: Specifies which fields must be filled before submitting the form.
+- `type`: Tells whether the data should be a number, email id, text string, etc.
+- `minLength`: Decides the minimum length for the text data string.
+- `maxLength`: Decides the maximum length for the text data string.
+
+Let's modify the form w.r.t. to these attributes:
+
+```html
+<!-- HTML Form with Built-in Validation -->
+<form action="/send-data-here" method="post">
+  <label for="roll">Roll Number</label>
+  <input
+    type="text"
+    id="roll"
+    name="roll"
+    required
+    minlength="10"
+    maxlength="20"
+  />
+  <label for="name">Name:</label>
+  <input type="text" id="name" name="name" required />
+  <button type="submit">Submit</button>
+</form>
+```
+
+With these validation checks in place, when a user tries to submit an empty field for Last Name, it gives an error that pops right in the form field. Similarly, a roll number can only be entered if it's 10-20 characters long.
```

Comment:
Where is the Last Name input? I see only Name

Reply:
Correct it's the Name. :)